src/core/unit.h

```diff
@@ -757,6 +757,11 @@ typedef struct UnitVTable {
 extern const UnitVTable * const unit_vtable[_UNIT_TYPE_MAX];
 
 static inline const UnitVTable* UNIT_VTABLE(const Unit *u) {
+        assert(u);
+
+        if (u->type < 0 || u->type >= _UNIT_TYPE_MAX)
```

Comment:
This helps with nothing though. UNIT_VTABLE()-> is still NULL ptr deref.

Reply:
Yes, but at least it doesn't go out of bounds in the array, so it's still better than status quo

Reply:
added check for vtable too in the gc path